src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -35,17 +36,22 @@ public class ClaimsTransformationMiddleware
 
         public async Task Invoke(HttpContext context)
         {
-            var handler = new ClaimsTransformationHandler(Options.Transformer, context);
+            var transform = Options.Transformer;
+            if (transform == null && Options.TransformerType != null)
+            {
+                transform = context.RequestServices.GetRequiredService(Options.TransformerType) as IClaimsTransformer;
```

Comment:
Um, that's a much bigger fish to fry, but I can spend a few cycles today(since it's moving day) with a prototype PR for cleaning up events in 1.2